### PR TITLE
Add board vid and pid in board list json output

### LINKED ISSUE #1069: Feature request: Can board list report the vid/pid found in the json output?
Related to arduino/arduino-cli#1068 :   Like the Leonardo, my Arduino-powered keyboards use a separate USB PID when they're in bootloader mode vs when they're in device mode. 

Because we want to avoid the possibility of unintentional or malicious flashing, our bootloader checks to see if a safety-interlock key is held down before accepting a sketch upload.  Otherwise, it boots into the user's sketch.

We have tooling that uses the device pid to determine whether a keyboard is in bootloader mode. Otherwise, it prompts the user to hold down the safety interlock key before doing the standard serial port tickle.

In the bad old days (with arduino-builder), we built our own logic to probe for serial ports by vid/pid pair. With arduino-cli, we no longer need to do that. Almost.

Right now, the json output from `arduino-cli board list`doesn't include the USB vendor-id and USB product-id reported back by serial-discovery.

If you folks would be willing to include the vid and pid in the json output from `board list`, I'd really appreciate it.  

(It might be useful to also include the board serial number, too, but I myself don't have an immediate need for it.)

Thanks!

### PR DESCRIPTION
**Pull request by silvanocerza**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?**

New information in an existing command output.

- **What is the current behavior?**

Boards `vid` and `pid` are not shown in `board list --format json` and are not included in the gRPC interface response.

* **What is the new behavior?**

`board list --format json` shows connected boards `vid` and `pid`, this also works when using the `--watch` flag.
This information is also included in the gRPC interface response.

Example output: 
```
[
  {
    "address": "/dev/ttyACM0",
    "protocol": "serial",
    "protocol_label": "Serial Port (USB)"
  },
  {
    "address": "/dev/ttyACM1",
    "protocol": "serial",
    "protocol_label": "Serial Port (USB)",
    "boards": [
      {
        "name": "Arduino Uno",
        "FQBN": "arduino:avr:uno",
        "VID": "0x2341",
        "PID": "0x0043"
      }
    ]
  }
]% 
```

- **Does this PR introduce a breaking change?**

No.

* **Other information**:

Fixes #1069.

---

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

### CODE REVIEW

**Review by per1234 — APPROVED** [review]
Looks great. Thanks Silvano!

**Comment by obra** [comment]
Thank you so much @silvanocerza!

File: commands/board/list.go [hunk 1]
```diff
@@ -118,6 +120,8 @@ func identify(pm *packagemanager.PackageManager, port *commands.BoardPort) ([]*r
 		boards = append(boards, &rpc.BoardListItem{
 			Name: board.Name(),
 			FQBN: board.FQBN(),
+			VID:  port.IdentificationPrefs.Get("vid"),
+			PID:  port.IdentificationPrefs.Get("pid"),
```
Comment (cmaglie) [inline commands/board/list.go:124]: Why just VID & PID, and not the whole `identificationPrefs`?
In this case we have only vid/pid but the pluggable discovery can provide much more information.

**Comment by abdullahmahboob** [comment]
Hey, this PR mentions that the `arduino-cli board list --format json` command returns `VID` and `PID` with the `--watch` flag as well, but I can't make it to return `VID` and `PID`. Here's how it works for me:

**Pull request merged**